app/models/group.rb

```diff
@@ -124,6 +126,15 @@ def add_leader_to_members
     members << leader unless members.include? leader
   end
 
+  # rubocop:disable RaiseArgs
+  def check_full_membership_validity
```

Comment:
why isn't this just the first validation? we can have it stop the callback chain and not validate further, i think, and that way we can customize the error message